File: 2015/1/main.py
with open('input.txt') as input_file:
    input = input_file.read()

floor = 0
count = 0
for i in input:
    count += 1
    if i == '(':
        floor += 1
    if i == ')':
        floor -= 1

    if floor == -1:
        print(count)
        break

File: 2015/1/main.go
package main

import (
	_ "embed"
	"fmt"
)

//go:embed input.txt
var input string

// 41 is )
// 40 is (

var count int
var iterations int

func main() {
	// Part 1
	for _, value := range input {
		if value == 40 {
			count = count + 1
		} else if value == 41 {
			count = count - 1
		}
	}
	fmt.Println("Part 1:")
	fmt.Println(count)

	// Part 2
	count = 0
	iterations = 0
	for _, value := range input {
		iterations += 1
		if value == 40 {
			count = count + 1
		} else if value == 41 {
			count = count - 1
		}
		if count == -1 {
			break
		}
	}
	fmt.Println("Part 2:")
	fmt.Println(iterations)
}
